nuxeo-features/nuxeo-automation/nuxeo-automation-core/src/test/java/org/nuxeo/ecm/automation/core/operations/document/CopyDocumentTest.java

```diff
@@ -0,0 +1,145 @@
+import org.nuxeo.ecm.core.api.DocumentModel;
+import org.nuxeo.ecm.core.api.DocumentModelList;
+import org.nuxeo.ecm.core.api.DocumentRef;
+import org.nuxeo.ecm.core.api.VersioningOption;
+import org.nuxeo.ecm.core.test.CoreFeature;
+import org.nuxeo.ecm.core.test.annotations.Granularity;
+import org.nuxeo.ecm.core.test.annotations.RepositoryConfig;
+import org.nuxeo.runtime.test.runner.Deploy;
+import org.nuxeo.runtime.test.runner.Features;
+import org.nuxeo.runtime.test.runner.FeaturesRunner;
+import org.nuxeo.runtime.test.runner.TransactionalFeature;
+
+/**
+ * @since 11.1
+ */
+@RunWith(FeaturesRunner.class)
+@Features(CoreFeature.class)
+@Deploy("org.nuxeo.ecm.automation.core")
+@Deploy("org.nuxeo.ecm.automation.core:OSGI-INF/copy-schema-test-contrib.xml")
+@RepositoryConfig(cleanup = Granularity.METHOD)
+public class CopyDocumentTest {
+
+    public static final String COPY_DOC_NAME = "CopyDoc";
+
+    public static final String ROOT = "/";
+
+    public static final String TARGET_PROPERTY_KEY = "target";
+
+    public static final String NAME_PROPERTY_KEY = "name";
+
+    public static final String SPECIAL_CHILD_DOC_NAME = "specialChildDoc";
+
+    public static final String REGULAR_UNDER_SPECIAL_CHILD_DOC_NAME = "regularUnderSpecialChildDoc";
+
+    public static final String REGULAR_CHILD_DOC_NAME = "regularChildDoc";
+
+    public static final String COMMENT_ROOT_TYPE = "CommentRoot";
+
+    public static final String FILE = "File";
+
+    @Inject
+    protected CoreSession session;
+
+    @Inject
+    protected AutomationService automationService;
+
+    @Inject
+    protected TransactionalFeature txFeature;
+
+    @Test
+    public void testCopyWithoutExcludedType() throws OperationException {
+        DocumentModel file = session.createDocumentModel(ROOT, "SourceDoc", FILE);
+        file = session.createDocument(file);
+
+        DocumentModel specialChildDoc = session.createDocumentModel(file.getPathAsString(), SPECIAL_CHILD_DOC_NAME,
+                COMMENT_ROOT_TYPE);
+        session.createDocument(specialChildDoc);
+
+        DocumentModel regularUnderSpecialDoc = session.createDocumentModel(specialChildDoc.getPathAsString(),
+                REGULAR_UNDER_SPECIAL_CHILD_DOC_NAME, FILE);
+        session.createDocument(regularUnderSpecialDoc);
+
+        DocumentModel regularChildDoc = session.createDocumentModel(file.getPathAsString(), REGULAR_CHILD_DOC_NAME,
+                FILE);
+        session.createDocument(regularChildDoc);
+
+        assertEquals(2, session.getChildren(file.getRef()).size());
+
+        // test nominal copy
+        try (OperationContext context = new OperationContext(session)) {
+            context.setInput(file);
+            Map<String, Serializable> params = new HashMap<>();
+            params.put(TARGET_PROPERTY_KEY, ROOT);
+            params.put(NAME_PROPERTY_KEY, COPY_DOC_NAME);
+            DocumentModel result = (DocumentModel) automationService.run(context, CopyDocument.ID, params);
```

Comment:
All this test class shouldn't be an Automation test at all. You should test the `CoreSession` methods only, and the tests should be alongside other copy tests in `TestSQLRepositoryAPI`.